sklearn/preprocessing/tests/test_data.py

```diff
@@ -851,6 +853,288 @@ def test_robust_scaler_iris_quantiles():
+                     [0, 0],
+                     [0, 1]])
+    assert_almost_equal(X_gt, X_trans.A)
+    assert_almost_equal(X_sparse.A, transformer.inverse_transform(X_trans).A)
+
+
+def test_quantile_transform_dense_toy():
+    X = np.array([[0, 25, 50, 75, 100],
+                  [2, 4, 6, 8, 10],
+                  [2.6, 4.1, 2.3, 9.5, 0.1]]).T
+
+    transformer = QuantileTransformer(n_quantiles=5)
+    transformer.fit(X)
+
+    X_trans = transformer.fit_transform(X)
+    assert_almost_equal(np.min(X_trans, axis=0), 0.)
+    assert_almost_equal(np.max(X_trans, axis=0), 1.)
+
+    X_test = np.array([
+        [-1,  1,  0],
+        [101, 11, 10],
+    ])
+    expected = np.array([
+        [0, 0, 0],
+        [1, 1, 1],
+    ])
+    assert_array_almost_equal(transformer.transform(X_test), expected)
+
+    X_trans_inv = transformer.inverse_transform(X_trans)
+    assert_array_almost_equal(X, X_trans_inv)
+
+    # test subsampling
+    # FIXME: there is not comparison for the moment
+    random_state = 42
+    transformer.set_params(**{'subsample': 3,
+                              'n_quantiles': 2,
+                              'random_state': random_state})
+    X_trans = transformer.fit_transform(X)
+
+
+def test_quantile_transform_sparse_toy():
+    X = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                  [2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                  [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+
+    X = sparse.csc_matrix(X)
+
+    transformer = QuantileTransformer(n_quantiles=10)
+    transformer.fit(X)
+
+    X_trans = transformer.fit_transform(X)
+    assert_array_almost_equal(np.min(X_trans.toarray(), axis=0), 0.)
+    assert_array_almost_equal(np.max(X_trans.toarray(), axis=0), 1.)
+
+    X_trans_inv = transformer.inverse_transform(X_trans)
+    assert_array_almost_equal(X.toarray(), X_trans_inv.toarray())
+
+    transformer_dense = QuantileTransformer(n_quantiles=10).fit(X.toarray())
+
+    X_trans = transformer_dense.transform(X)
+    assert_array_almost_equal(np.min(X_trans.toarray(), axis=0), 0.)
+    assert_array_almost_equal(np.max(X_trans.toarray(), axis=0), 1.)
+
+    X_trans_inv = transformer_dense.inverse_transform(X_trans)
+    assert_array_almost_equal(X.toarray(), X_trans_inv.toarray())
+
+    # test subsampling
+    # FIXME: there is not comparison for the moment
+    random_state = 42
+    transformer.set_params(**{'subsample': 3,
+                              'n_quantiles': 2,
+                              'random_state': random_state})
+    X_trans = transformer.fit_transform(X)
+
+
+def test_quantile_transform_axis1():
+    X = np.array([[0, 25, 50, 75, 100],
+                  [2, 4, 6, 8, 10],
+                  [2.6, 4.1, 2.3, 9.5, 0.1]])
+
+    X_trans_a0 = quantile_transform(X.T, axis=0, n_quantiles=5)
+    X_trans_a1 = quantile_transform(X, axis=1, n_quantiles=5)
+    assert_array_almost_equal(X_trans_a0, X_trans_a1.T)
+
+
+def test_qunatile_transform_bounds():
```

Comment:
I don't get what you're testing here. I'd thought you might be checking the behaviour where the transformed data includes values beyond the range of the fit data. But that doesn't seem to be the case.